specification/web/resource-manager/Microsoft.DomainRegistration/stable/2022-09-01/Domains.json

```diff
@@ -0,0 +1,1702 @@
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/DomainCollection"
+            }
+          },
+          "default": {
+            "description": "App Service error response.",
+            "schema": {
+              "$ref": "../../../Microsoft.Web/stable/2022-09-01/CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List domains by resource group": {
+            "$ref": "./examples/ListDomainsByResourceGroup.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DomainRegistration/domains/{domainName}": {
+      "get": {
+        "tags": [
+          "Domains"
+        ],
+        "summary": "Get a domain.",
+        "description": "Description for Get a domain.",
+        "operationId": "Domains_Get",
+        "parameters": [
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "name": "domainName",
+            "in": "path",
+            "description": "Name of the domain.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/Domain"
+            }
+          },
+          "default": {
+            "description": "App Service error response.",
+            "schema": {
+              "$ref": "../../../Microsoft.Web/stable/2022-09-01/CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get Domain": {
+            "$ref": "./examples/GetDomain.json"
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "Domains"
+        ],
+        "summary": "Creates or updates a domain.",
+        "description": "Description for Creates or updates a domain.",
+        "operationId": "Domains_CreateOrUpdate",
+        "parameters": [
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "name": "domainName",
+            "in": "path",
+            "description": "Name of the domain.",
+            "required": true,
+            "type": "string",
+            "pattern": "[a-zA-Z0-9][a-zA-Z0-9\\.-]+"
+          },
+          {
+            "name": "domain",
+            "in": "body",
+            "description": "Domain registration information.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/Domain"
+            }
+          },
```

Comment:
Considering factoring these as single definitions in the "Parameters" section of your spec, and using $ref to them everywhere else, easier to review and maintain